File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/UserService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.UserDTO;
import com.hexaware.automobileinsurance.model.User;

public interface UserService {

    User registerUser(UserDTO dto);

    User getUserById(Integer userId);

    List<User> getAllUsers();

    User updateUser(Integer userId, UserDTO dto);

    void deleteUser(Integer userId);

    // Generates a new temporary password for the given email, saves it
    // (hashed), and emails it to the user. Used by the public
    // /auth/forgot-password endpoint.
    void resetPassword(String email);
}

File: automobile-insurance-frontend/src/services/UserService.js
import api from "../api/api";

const UserService = {

    getAllUsers() {
        return api.get("/users");
    },

    getUserById(userId) {
        return api.get(`/users/${userId}`);
    },

    registerUser(user) {
        return api.post("/users", user);
    },

    updateUser(userId, user) {
        return api.put(`/users/${userId}`, user);
    },

    deleteUser(userId) {
        return api.delete(`/users/${userId}`);
    },getMyProfile() {
        return api.get("/users/me");
    },

    updateMyProfile(user) {
        return api.put("/users/me", user);
    }

};

export default UserService;